crwatcher/watcher.go

```diff
@@ -114,20 +116,47 @@ func (cw *CRWatcher) setupHandler(con ResourceController) {
 	cw.handler = cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			r := obj.(*unstructured.Unstructured)
+			if cw.Config.Filter != "" && !cw.hasFilter(r) {
```

Comment:
Could we move the check for the filter into the method we call. Then update it to work like the documentation says it will work? Maybe make this somehting like 
``` 
if cw.clearsFilter(r) {
    ..do stuff..
}
```
Then clearsFilter returns true if the resource is not filtered out. I think this helps a lot with readability not having to have `!hasFilter`

Reply:
Done